lightwood/mixers/nn.py

```diff
@@ -150,22 +150,29 @@ def fit(self, train_ds, test_ds, stop_training_after_seconds=None):
             stop_training_after_seconds = self.stop_training_after_seconds
 
         self.fit_data_source(train_ds)
+        self.fit_data_source(test_ds)
+
+        assert train_ds.transformer is test_ds.transformer
+        assert train_ds.encoders is test_ds.encoders
```

Comment:
Why are these asserts needed here ?

Seems to me like these should be part of the tests for nn.py, not part of the logic here.

Reply:
because it's assumed that test_ds is the same as train_ds (same columns, encoders, transformer) but with different rows, it will raise assertion error if someone attempts to use different datasources for train and test